elements/lisk-transactions/src/10_delegate_transaction.ts

```diff
@@ -92,8 +100,31 @@ export class DelegateTransaction extends BaseTransaction {
 		const errors: TransactionError[] = [];
 		const sender = await store.account.get(this.senderId);
 
-		const usernameExists = store.account.find(
-			(account: Account) => account.username === this.asset.username,
+		// Data format for the registered delegates
+		// chain:usernames => { registeredDelegates: { username, address }[] }
+		const registeredDelegatesStr = await store.chain.get(
+			DELEGATE_USERNAME_CHAIN_KEY,
+		);
+		const registeredDelegates = registeredDelegatesStr
+			? (JSON.parse(registeredDelegatesStr) as ChainUsernames)
+			: { registeredDelegates: [] };
+		const usernameExists = registeredDelegates.registeredDelegates.find(
+			delegate => delegate.username === this.asset.username,
+		);
+		const updatedRegisteredDelegates = usernameExists
+			? { ...registeredDelegates }
+			: {
+					registeredDelegates: [
+						...registeredDelegates.registeredDelegates,
```

Comment:
instead of spreading, push should be fine.
Also, it's better to be sorted by `username` or `address` to make it deterministic